doajtest/unit/event_consumers/test_application_publisher_revision_notify.py

```diff
@@ -0,0 +1,65 @@
+from portality import models
+from portality import constants
+from portality.bll import exceptions
+from doajtest.helpers import DoajTestCase
+from doajtest.fixtures import ApplicationFixtureFactory
+import time
+
+from portality.events.consumers.application_publisher_revision_notify import ApplicationPublisherRevisionNotify
+
+
+class TestApplicationPublisherRevisionNotify(DoajTestCase):
+    def setUp(self):
+        super(TestApplicationPublisherRevisionNotify, self).setUp()
+
+    def tearDown(self):
+        super(TestApplicationPublisherRevisionNotify, self).tearDown()
+
+    def test_consumes(self):
+        source = ApplicationFixtureFactory.make_application_source()
+
+        event = models.Event(constants.EVENT_APPLICATION_STATUS, context={"application": "abcd", "old_status": "in progress", "new_status": "revisions_required"})
+        assert ApplicationPublisherRevisionNotify.consumes(event)
+
+        event = models.Event(constants.EVENT_APPLICATION_STATUS,
+                             context={"application": "abcd", "old_status": "revisions_required", "new_status": "revisions_required"})
+        assert not ApplicationPublisherRevisionNotify.consumes(event)
+
+        event = models.Event("test:event", context={"application" : "abcd"})
+        assert not ApplicationPublisherRevisionNotify.consumes(event)
+
+        event = models.Event(constants.EVENT_APPLICATION_STATUS)
+        assert not ApplicationPublisherRevisionNotify.consumes(event)
+
+    def test_consume_success(self):
+        self._make_and_push_test_context("/")
+
+        source = ApplicationFixtureFactory.make_application_source()
+        app = models.Application(**source)
+        app.save()
+
+        acc = models.Account()
+        acc.set_id("publisher")
+        acc.set_email("test@example.com")
+        acc.save()
+
+        event = models.Event(constants.EVENT_APPLICATION_STATUS, context={"application": "abcdefghijk", "old_status": "in progress", "new_status": "revisions_required"})
```

Comment:
the `application` context argument needs to be the core data structure for an actual application to which the event occurred.  You can do

```
source = ApplicationFixtureFactory.make_application_source()
app = models.Application(**source)
event = models.Event(constants.EVENT_APPLICATION_STATUS, context={"application": app.data, "old_status": "in progress", "new_status": "revisions_required"})
```